bip-signet.mediawiki

```diff
@@ -0,0 +1,85 @@
+
+== Motivation ==
+
+Testnet is a great place to try out new things without risking real money, but it is notoriously unreliable. Huge block reorgs, long gaps in between blocks being mined or sudden bursts of blocks in rapid succession mean that realistic testing of software, especially involving multiple independent parties running software over an extended period of time, becomes infeasible in practice.
+
+A new type of test network would be more suitable for integration testing by organizations such as exchanges, or testing of next generation Layer-2 protocols like Eltoo or sidechain pegs. The goal is not to be perfectly reliable but rather to have a predictable amount of unreliability. You want a test network to behave like mainnet (i.e. no thousands of block reorgs) while also making it easier to trigger expected but rare events like a 6-block reorg. Regtest is not suitable for longer-term scenarios involving multiple independent parties because creating blocks costs nothing, so any party can completely control the test network.
+
+
+== Specification ==
+
+A new type of network ("signet"), which takes an additional consensus parameter called the challenge (scriptPubKey). The challenge can be a simple pubkey (P2PKH style), or a k-of-n multisig, or any other script you would want.
+
+The witness commitment of the coinbase transaction is extended to include a secondary commitment (the signature/solution):
+
+    1-4 bytes - Push the following (x + 4) bytes
+    4 bytes - Signet header (0xecc7daa2)
+    x bytes - Solution (sigScript)
+
+Any push operations that do not start with the 4 byte signet header are ignored. Multiple push operations with the 4 byte signet header are ignored except for the first entry.
+
+Any signature operations contained within the challenge use SHA256d(modifiedBlockHash), i.e. the double-SHA256 digest of the following data as the sighash:
+
+{|class="wikitable" style="text-align: center;"
+|-
+!Type
+!Size
+!Name
+|-
+|Int32||4||nVersion
+|-
+|Uint256||32||hashPrevBlock
+|-
+|Uint256||32||modifiedMerkleRoot
+|-
+|Uint32||4||nTime
+|-
+|Uint32||4||nBits
+|}
+
+The <code>modifiedMerkleRoot</code> hash is obtained by generating the merkle root of the block transactions, with the coinbase witness commitment as is, without the signet extension. This means the merkle root of the block is different from the merkle root in the signet commitment, but in return, the block nonce value is the only component that the signet signature does not commit to. When grinding proof of work, the extended nonce cannot be used as it would invalidate the signature. Instead, simply resigning the same (or an updated) block will give a new search space.
+
+A block is considered fully validated if the above commitment is found, and its solution is valid. This verification should be done directly before or after the witness commitment verification.
```

Comment:
You cannot do this verification at the same place that you do proof of work verification, because the data necessary in the two cases is different. For POW validation you only need the block header, whereas the block signature requires the block in its entirety, including all of the transactions.

There is no real advantage to doing it before vs after witness validation, but it should be considered an integral *part* of witness validation; it resides in the witness commitment, and it requires approximately the same data to operate.

Reply:
well, by "the same place" I guess I meant one after the other, not necessarily the same function.
I get that both segwit and signet validation require the coinbase transaction whereas pow validation doesn't. But, sorry, I still don't get why signet validation should be done near segwit validation.

Reply:
Where would you prefer it be done? It seems natural to do it around witness commitment check, since the data is there already.

Reply:
I removed "should" and turned this into a recommendation.